### PR TITLE
feat(logging): Make logs public and print version

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
Two-fold pull request:
- Windows Agent:
    - The logs are moved to the public directory @ $HOME/.ubuntupro/log
    - The logs print the version
- WSL Pro Service:
    - The logs print the version

### CODE REVIEW

**Comment by EduardGomezEscandell** [comment]
Regarding codecov: these are changes to the main file, which cannot be tested.

**Comment by EduardGomezEscandell** [comment]
Force-pushed to fix merge conflicts.

File: windows-agent/cmd/ubuntu-pro-agent/agent/agent.go [hunk 1]
```diff
@@ -229,3 +201,50 @@ func (a App) RootCmd() cobra.Command {
...
+	return publicDir, nil
+}
+
+// publicDir is a wrapper around PublicDir to allow overriding its path with an option.
+func (a *App) publicDir(opts options) (string, error) {
+	if opts.publicDir == "" {
+		return a.PublicDir()
+	}
+
+	if err := os.MkdirAll(opts.publicDir, 0600); err != nil {
```
Comment (didrocks) [inline windows-agent/cmd/ubuntu-pro-agent/agent/agent.go:227]: I am a bit uneasy that both `PublicDir()` and `publicDir()` create the directory. Can we maybe simplify in one of those which takes an option?

File: windows-agent/cmd/ubuntu-pro-agent/agent/agent_test.go [hunk 2]
```diff
@@ -181,6 +182,50 @@ func TestAppGetRootCmd(t *testing.T) {
...
+	// Not parallel because we modify the environment
+
+	testCases := map[string]struct {
+		emptyEnv bool
+		badPath  bool
+
+		wantErr bool
+	}{
+		"Success providing a public directory":               {},
+		"Error when %UserProfile% is empty":                  {emptyEnv: true, wantErr: true},
```
Comment (didrocks) [inline windows-agent/cmd/ubuntu-pro-agent/agent/agent_test.go:195]: I know there are 3 cases but still separating the success/error cases by a blank line sounds like a better way if we add future parameters to it :)

**Review by didrocks — CHANGES_REQUESTED** [review]
My only concern is this `publicDir` vs `PublicDir` on the agent where we may be able to consolidate them in one (with a variadic optional parameter). That way, we would avoid having `MkdirAll` in 2 places.

Otherwise, great! Thanks for updating the path to separate user/sysadmin data vs config :)

File: windows-agent/cmd/ubuntu-pro-agent/agent/agent.go (hunk 1 above)
Comment (EduardGomezEscandell) [reply windows-agent/cmd/ubuntu-pro-agent/agent/agent.go:227]: I just didn't want the public interface to take this optional argument as it is a bit redundant. But I guess it's fine.

File: windows-agent/cmd/ubuntu-pro-agent/agent/agent_test.go (hunk 2 above)
Comment (EduardGomezEscandell) [reply windows-agent/cmd/ubuntu-pro-agent/agent/agent_test.go:195]: Yep, my bad.

File: windows-agent/cmd/ubuntu-pro-agent/agent/agent.go (hunk 1 above)
Comment (EduardGomezEscandell) [reply windows-agent/cmd/ubuntu-pro-agent/agent/agent.go:222]: In the end I inverted the logic (PublicDir calls publicDir).

I just did not want the App interface to have method `PublicDir(string)`. Feels like exposing its internals a bit.

**Pull request merged**